libs/model/src/policies/CommunityIndexer.policy.ts

```diff
@@ -0,0 +1,234 @@
+import { config } from '@hicommonwealth/model';
+import * as schemas from '@hicommonwealth/schemas';
+import {
+  ClankerToken,
+  CommunityIndexer as CommunityIndexerSchema,
+  EventNames,
+  EventPairs,
+  events,
+} from '@hicommonwealth/schemas';
+import { ChainBase, ChainType } from '@hicommonwealth/shared';
+import axios from 'axios';
+import lo from 'lodash';
+import moment from 'moment';
+import { Literal } from 'sequelize/lib/utils';
+import { z } from 'zod';
+import { CreateCommunity } from '../community';
+import { models, sequelize } from '../database';
+import { systemActor } from '../middleware';
+import { mustExist } from '../middleware/guards';
+import { emitEvent } from '../utils';
+
+const log = logger(import.meta);
+
+const inputs = {
+  CommunityIndexerTimerTicked: events.CommunityIndexerTimerTicked,
+  ClankerTokenFound: events.ClankerTokenFound,
+};
+
+type CommunityIndexerStatus = z.infer<typeof CommunityIndexerSchema>['status'];
+
+async function setIndexerStatus(
+  indexerId: string,
+  status: CommunityIndexerStatus,
+) {
+  const toUpdate: {
+    status: CommunityIndexerStatus;
+    last_checked?: Literal;
+  } = {
+    status,
+  };
+  if (status === 'pending') {
+    toUpdate.last_checked = sequelize.literal('NOW()');
+  }
+  return await models.CommunityIndexer.update(toUpdate, {
+    where: {
+      id: indexerId,
+    },
+  });
+}
+
+const sleepAsync = (delay: number) =>
+  new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(null);
+    }, delay);
+  });
+
+async function fetchClankerTokens(
+  cutoffDate: Date,
+  onPage: (tokens: Array<z.infer<typeof ClankerToken>>) => Promise<void>,
+) {
+  let numRetries = 3;
+  let backoff = 5000;
+  let pageNum = 1;
+
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    const url = `https://www.clanker.world/api/tokens?sort=desc&page=${pageNum}&pair=all&partner=all&presale=all`;
+    log.debug(`fetching: ${url}`);
+
+    try {
+      const res = await axios.get<{
+        data: Array<z.infer<typeof ClankerToken>>;
+      }>(url);
+
+      // Stop fetching if no more tokens
+      if (res.data.data.length === 0) {
+        log.debug('No more tokens found');
+        break;
+      }
+
+      // Filter tokens by cutoffDate
+      const validTokens = res.data.data.filter((t) =>
+        moment(t.created_at).isAfter(cutoffDate),
+      );
+
+      if (validTokens.length > 0) {
+        await onPage(validTokens);
+      }
+
+      // Check if the oldest token is older than the cutoffDate
+      const oldestToken = res.data.data[res.data.data.length - 1];
+      if (moment(oldestToken.created_at).isBefore(cutoffDate)) {
+        break;
+      }
+
+      // Reset retries and move to the next page
+      numRetries = 3;
+      backoff = 5000;
+      pageNum++;
+    } catch (err: any) {
+      // Retry if rated limited
+      if (err.response?.status === 429) {
+        log.warn('Encountered clanker API rate limiting');
```

Comment:
we should probably create a generic util for this concept - retries with backoff